addons/website/controllers/main.py

```diff
@@ -339,16 +326,15 @@ def multi_render(self, ids_or_xml_ids, values=None):
         '/website/action/<path_or_xml_id_or_id>/<path:path>',
         ], type='http', auth="public", website=True)
     def actions_server(self, path_or_xml_id_or_id, **post):
-        cr, uid, context = request.cr, request.uid, request.context
         res, action_id, action = None, None, None
-        ServerActions = request.registry['ir.actions.server']
+        ServerActions = request.env['ir.actions.server']
 
         # find the action_id: either an xml_id, the path, or an ID
         if isinstance(path_or_xml_id_or_id, basestring) and '.' in path_or_xml_id_or_id:
-            action_id = request.registry['ir.model.data'].xmlid_to_res_id(request.cr, request.uid, path_or_xml_id_or_id, raise_if_not_found=False)
+            action_id = request.env.ref(path_or_xml_id_or_id, raise_if_not_found=False).id
```

Comment:
use browse record.

after that, probability to remove code from line:337 to line:348